ide-common/src/main/kotlin/org/digma/intellij/plugin/editor/PaginationUtil.kt

```diff
@@ -0,0 +1,36 @@
+package org.digma.intellij.plugin.editor
+
+import java.util.concurrent.ConcurrentHashMap
+
+private val insightsPaginationInfo: ConcurrentHashMap<String, ConcurrentHashMap<String, Int>> = ConcurrentHashMap()
+
+fun addInsightPaginationInfo(focusedDocumentName: String, uniqueInsightId: String, activePage: Int) {
+    if (activePage > 1) {
+        val insightsPaginationInfosMapForActualFile = insightsPaginationInfo.get(focusedDocumentName)
+        if (insightsPaginationInfosMapForActualFile != null) {
+            insightsPaginationInfosMapForActualFile[uniqueInsightId] = activePage
+            insightsPaginationInfo[focusedDocumentName] = insightsPaginationInfosMapForActualFile
+        } else {
+            val filesAndInsightsPaginationInfosMap: ConcurrentHashMap<String, Int> = ConcurrentHashMap()
+            filesAndInsightsPaginationInfosMap[uniqueInsightId] = activePage
+            insightsPaginationInfo[focusedDocumentName] = filesAndInsightsPaginationInfosMap
+        }
+    } else if (insightsPaginationInfo.contains(uniqueInsightId)) {
+        // remove pagination info for insights which had some page selected before
+        insightsPaginationInfo.remove(uniqueInsightId)
+    }
+}
+
+fun getInsightPaginationInfo(uniqueInsightId: String): Int? {
```

Comment:
Shouldn't it take into account the `focusedDocumentName` ?

Reply:
thank you for useful advice!